src/main/java/sirius/biz/translations/BasicTranslations.java

```diff
@@ -0,0 +1,177 @@
+
+    /**
+     * Deletes the translation for the given field and language
+     *
+     * @param field {@link Mapping} of the translated field
+     * @param lang  code of the language to be deleted
+     */
+    public abstract void deleteText(Mapping field, String lang);
+
+    /**
+     * Deletes all translations for the given field
+     *
+     * @param field {@link Mapping} of the translated field
+     */
+    public abstract void deleteAllTexts(Mapping field);
+
+    /**
+     * Checks, if the given language is supported by the system and returns the translation for the given parameters,
+     * or creates a new one if none exists.
+     *
+     * @param field {@link Mapping} of the translated field
+     * @param lang  language code
+     * @param text  translated text for the given language
+     * @return the translation entity matching the database in use (Mongo or SQL), null if language not supported
+     */
+    protected abstract T findOrCreateTranslation(Mapping field, String lang, String text);
+
+    /**
+     * Gets the translated text for the given field and language.
+     *
+     * @param field {@link Mapping} of the translated field
+     * @param lang  language code
+     * @return {@link Optional} with translated text for the given language, or empty if none is found
+     */
+    public Optional<String> getText(Mapping field, String lang) {
+        if (!isSupportedLanguage(lang)) {
+            return Optional.empty();
+        }
+
+        Optional<T> translation = fetchTranslation(field, lang);
+        return translation.map(t -> t.getTranslationData().getText());
+    }
+
+    /**
+     * Returns the translated text for the given field and language. Uses fallback language if no translation is found.
+     * If the fallback also has no translation, the default text from the owner entity is returned.
+     *
+     * @param field        {@link Mapping} of the translated field
+     * @param lang         language code
+     * @param fallbackLang language code for fallback
+     * @return translated text for the given language, fallback, or the default text from owner entity
+     */
+    public String getRequiredText(Mapping field, String lang, String fallbackLang) {
+        Optional<String> text = getText(field, lang);
+        if (!text.isPresent()) {
+            text = getText(field, fallbackLang);
+            if (!text.isPresent()) {
+                return fetchDefaultText(field);
+            }
+        }
+        return text.get();
+    }
+
+    /**
+     * Fetches the default text by querying {@link sirius.db.mixing.FieldLookupCache} for the owner entity's given field.
+     *
+     * @param field {@link Mapping field} to fetch the the text from
+     * @return the text as String, or null if lookup was unsuccessful
+     */
+    protected String fetchDefaultText(Mapping field) {
+        return fieldLookupCache.lookup(owner.getClass(), owner.getId(), field).asString();
```

Comment:
why not directly read from the owner? owner.getDescriptor.getProperty(field).getValue(owner) or the like ? maybe check if the value is fetched and else fallback to the fieldLookupCache otherwise(??)